engines/python/src/main/java/ai/djl/python/engine/PyModel.java

```diff
@@ -149,13 +150,20 @@ public void load(Path modelPath, String prefix, Map<String, ?> options) throws I
             pyEnv.setFailOnInitialize(false);
         }
 
+        Boolean secureModeEnabled = isCustomEntryPointDisallowed();
+
         if (entryPoint == null) {
             entryPoint = Utils.getenv("DJL_ENTRY_POINT");
             if (entryPoint == null) {
                 Path modelFile = findModelFile(prefix);
                 String features = Utils.getEnvOrSystemProperty("SERVING_FEATURES");
                 // find default entryPoint
-                if (modelFile != null) {
+                if (secureModeEnabled) {
```

Comment:
We don't need check everywhere, we can check if the final `entryPoint.startsWith("djl_python.")`